### DISCUSSION TITLE
New feature from cs2 game coordinator

### DISCUSSION THREAD

**Discussion by OliverTrust** [Gobot1234/steam.py#353] [Q&A]
Is any way to choose weekly level reward?

**Comment by Gobot1234** [comment]
I haven't implemented anything in the module outside of the original game so no not currently. If you want to figure out how to make it work then use steam kits nethook and figure out how it works.

**Reply by OliverTrust** [reply]
It is not "outside" of the original game. [CMsgGCCstrike15_v2_ClientRedeemFreeReward](https://github.com/SteamDatabase/Protobufs/blob/cd96974acf68ad3781f07fdc5e8a52d8cb7e2c49/csgo/cstrike15_gcmessages.proto#L750-L754) In new proto CS2

**Reply by Gobot1234** [reply]
Yes I'm saying I haven't implemented features that weren't in just bog standard CSGO, I haven't looked at CS2 or anything

**Reply by ALPhaHoai** [reply]
hi, how about now

**Reply by Gobot1234** [reply]
It's now implemented

**Reply by ALPhaHoai** [reply]
> It's now implemented

could you give me name of the function and did you use CMsgGCCstrike15_v2_ClientRedeemFreeReward to redeem?

**Reply by Gobot1234** [reply]
Oh apologies no this hasn't been included, wouldn't be difficult to implement

**Reply by ALPhaHoai** [reply]
I'm not familiar with python, could you give me advice to implement this

**Reply by Gobot1234** [reply]
> I'm not familiar with python, could you give me advice to implement this

Thank you for the interest but it's already done (I think it needs it be in a release though), maybe take a look at the project board if you want things to do.

**Comment by Gobot1234** [comment]
The method FWIW is csgo.Client.redeem_weekly_reward

**Reply by ALPhaHoai** [reply]
thank you so much

**Reply by PONYAWKA** [reply]
@Gobot1234 is it still in place?  am cannot find this function in code

**Reply by Gobot1234** [reply]
Yes, you might need to update your version

> @Gobot1234 is it still in place? am cannot find this function in code

**Reply by PONYAWKA** [reply]
You are totaly right, thank you
But what about getting info about items in on_weekly_reward, now its returns me and id and position
There extrastep to get info about item ? Or smth wrong in parsing item payload ?

**Reply by Gobot1234** [reply]
Currently unless you know the details of the items you are requesting you can't get them I don't think

**Reply by PONYAWKA** [reply]
but am actualy can, in update_backpack method, that am currently patching you are parsing all requred info, but then is some reason you doing utils.update_class(gc_item, _HackBaseItem()), and somehow in my case its casting object to have only item_id and position

**Reply by Gobot1234** [reply]
That sounds unrelated but I'm not really sure what you're saying sorry

**Reply by PONYAWKA** [reply]
Long story shot am patched it for my needs
But it still not working for me as client.redeem_weekly_reward with correct data (ids, time, and balance) not getting items
Do you have examples probably ? Or know with what this issue related 
Already a week looking for solution heh

**Reply by ifokeev** [reply]
@PONYAWKA It's not working for me either. Have you found a solution? cc @Gobot1234
